design-proposals/klip-47-add-source-stream-table-semantic.md

```diff
@@ -0,0 +1,122 @@
+# KLIP 47 - Add Source Stream/Table Semantic 
+
+**Author**: Boyang Chen (@boyang) | 
+**Release Target**: TBD | 
+**Status**: _In Discussion_ | 
+**Discussion**: https://github.com/confluentinc/ksql/pull/7474
+
+**tl;dr:** _Add support for source stream/table. This will give users a read-only view of data that 
+are populated by the upstream Kafka topic, making the data ownership more clear._
+           
+## Motivation and background
+
+Users of KsqlDB have no clear ownership definition around tables or streams they created. In a “normal” database, 
+when a user creates a table the DB creates the corresponding files on disk and owns the table. 
+User may write data into those tables via `INSERT INTO` statements and/or modify/delete existing data 
+via `UPDATE` and `DELETE` statements. However, the table will be not changed by any other means. 
+
+For ksqlDB, the setup is quite differently. Specifically, user could create the database based on 
+the topics stored in Kafka as the source of truth. The defined database could be mutable by 
+using `INSERT` command to populate data into it, which will then pollute the original data source.
+Secondly, when calling `CREATE TABLE`, the actual data is neither materialized nor 
+ready for interactive queries to access. To address these gaps, we would like to add:
+
+* Syntax to add read-only stream/table as `CREATE SOURCE STREAM/TABLE` to make the data source as read-only. Any 
+insertion command will be rejected, with an example error response as:
+
+```
+Error: insertion into source stream/table is not allowed
+```   
+
+* In addition, the `CREATE SOURCE TABLE` will materialize the input source topic into a stream 
+instance which would generate a RocksDB instance for interactive queries.
```

Comment:
"stream instance" might be confusing -- I understand that you refer to Kafka Stream, but not everybody might have enough context.

It might be worth to point out, that `CREATE SOURCE TABLE` is not just a metadata operation like `CREATE TABLE`, `CREATE STREAM`, and `CREATE SOURCE STREAM`, but effectively deploys a "persistent query" (for a lack of a better term) in the ksqlDB servers.